Source/EasyNetQ/Producer/ClientCommandDispatcherSingleton.cs

```diff
@@ -101,13 +104,15 @@ private void StartDispatcherThread(ConnectionConfiguration configuration)
                         var channelAction = queue.Take(cancellation.Token);
                         channelAction();
                     }
-                    catch (OperationCanceledException)
+                    catch (OperationCanceledException) when (cancellation.IsCancellationRequested)
```

Comment:
This change doesn't make any sense because in both cases thread stops its work (:

Reply:
Oops, agree. I did not notice that there is no
```c#
 catch (Exception exception)
                    {
                        logger.ErrorException(string.Empty, exception);
                    }
``` 
Why so?

Reply:
The reason is that all exceptions seem to be handled [during action creation](https://github.com/EasyNetQ/EasyNetQ/blob/master/Source/EasyNetQ/Producer/ClientCommandDispatcherSingleton.cs#L49)

Reply:
:facepalm: